.github/scripts/generate-checks-pages.py

```
@@ -164,35 +162,61 @@ def write_to_file(self, file_path: Path, overwrite: bool = False):
-        output_file_path = output_folder.joinpath(f"Check-{html_file_name}.md")
-
-        print(f"Generating {output_file_path}")
-
-        converter = HTMLToMarkdown(html_file_path)
-        converter.convert_to_markdown()
-        converter.write_to_file(output_file_path, True)
-
-        java_file = java_checks_folder.joinpath(f"{html_file_name}Check.java")
-        if java_file.exists():
-            converter = JavaToMarkdown(java_file)
+    index_writer.write_to_file("# Available checks\n", True)
+
+    CHECK_TYPES = {
+        "Magik checks": (
+            Path("magik-checks/src/main/java/nl/ramsolutions/sw/checks/magik"),
+            Path("magik-checks/src/main/resources/nl/ramsolutions/sw/sonar/l10n/magik/rules")
+        ),
+        "Magik typed checks": (
+            Path("magik-checks/src/main/java/nl/ramsolutions/sw/checks/magiktyped"),
+            Path("magik-checks/src/main/resources/nl/ramsolutions/sw/sonar/l10n/magiktyped/rules")
+        ),
+        "module.def checks": (
+            Path("magik-checks/src/main/java/nl/ramsolutions/sw/checks/moduledef"),
+            Path("magik-checks/src/main/resources/nl/ramsolutions/sw/sonar/l10n/moduledef/rules")
+        ),
+        "product.def checks": (
+            Path("magik-checks/src/main/java/nl/ramsolutions/sw/checks/productdef"),
+            Path("magik-checks/src/main/resources/nl/ramsolutions/sw/sonar/l10n/productdef/rules")
+        ),
+    }
+
+    for check_type, (java_checks_folder, sonar_rules_folder) in CHECK_TYPES.items():
+        index_writer.write_to_file(f"\n## {check_type}\n\n")
+
+        for html_file_path in sorted(sonar_rules_folder.glob("*.html")):
+            file_name = html_file_path.stem
+            output_file_path = output_folder.joinpath(f"Check-{file_name}.md")
+
+            print(f"Generating {output_file_path}")
+
+            json_file_path = sonar_rules_folder / f"{file_name}.json"
+            if json_file_path.exists():
```

Don't test if it exists, just let it crash when the file is expected to be there but isn't.